api/controllers/UserController.js

```diff
@@ -253,41 +253,53 @@ module.exports = {
       console.log(err);
       return res.status(500).json(err);
     }
-    Reddit.getBothFlairs(req.user.redToken, req.params.username).then(function (flairs) {
-      var logged_fcs = user ? user.loggedFriendCodes : [];
+    var flairs;
+    flairs = await Reddit.getBothFlairs(req.user.redToken, req.params.username).catch(function (err) {
```

Comment:
Do a try catch instead of catching the promise.

So:

try{ 
flairs = await Reddit.getBothFlairs(
} catch (err) {
if err.statusCode !== 403 {
throw err;
}
}